### DISCUSSION TITLE
What is the semantics of closing a scene tab?

### DISCUSSION THREAD

**Discussion by voroninp** [godotengine/godot-proposals#8373] [Editor]
I have an active scene:

1.
![image](https://github.com/godotengine/godot-proposals/assets/9202028/dd545fe8-9241-4a58-9ae6-8402b295b15c)

When I try to close the tab for `main.tscn` the screen changes to this one:

2.
![image](https://github.com/godotengine/godot-proposals/assets/9202028/71d9c494-f1c6-4a7e-87e0-89923f007627)

However, withing this tab I can do `File -> Close All` and then I get an empty tab:

3.
![image](https://github.com/godotengine/godot-proposals/assets/9202028/f7e7ab41-41cf-4aad-b360-66ff0515a3b0)

Probably, I have the wrong mental model of the screen areas and how they depend on each other. But can anyone tell me what that interim (2) state does and how the list of scripts depend on the scene?

![image](https://github.com/godotengine/godot-proposals/assets/9202028/7e3b3906-2044-4848-ae02-b51d8837602a)

**Comment by AThousandShips** [comment]
This is a common confusion, but the script tab is not related to the scenes tab, closing the scene doesn't close the script even if it is associated, because that would be confusing

However closing *all* files closes all files

**Reply by voroninp** [reply]
And what does the scene tab affect in the central area then?
![image](https://github.com/godotengine/godot-proposals/assets/9202028/fd39d9e6-d83b-4779-8c8a-76d19b4b0107)

**Reply by AThousandShips** [reply]
How do you mean? It has the scene editing, it can also open or select the correct script, but the script editor is separate

**Reply by voroninp** [reply]
When I switch between scene tabs player/hud I see no changes in the area on the screenshot.
`Scene` and `Inspector` pane do change.

**Reply by AThousandShips** [reply]
Yes, that is what I said, the script editor isn't connected, it is for scripts, not the scene

I am not sure what is the confusing part here?

**Reply by voroninp** [reply]
Exactly this =)
I mean the proximity creates an impression that it contains the script area.
![image](https://github.com/godotengine/godot-proposals/assets/9202028/a6e2a723-c354-44aa-b08b-7c9d10f6d6c0)

IMO, there should be a noticeable gap between the scenes tab row and everything else:
![image](https://github.com/godotengine/godot-proposals/assets/9202028/e38df748-9e72-4d2e-a06e-f867cff3da8b)

**Reply by AThousandShips** [reply]
This is simply something you need to get used to, it work, it makes sense, and some times user interface isn't intuitive to everyone, and then it's important to adapt and learn, it seems to make sense to most people and changing it would create a lot of confusion

**Reply by theraot** [reply]
@voroninp The script editor might use the currently edited scene for a few things: find out signal connections and generating code when dropping nodes. However, one is not conceptually inside the other, despite what the UI suggests.

As pointed out, the scripts are not inside the current scene. Similarly the scenes are not conceptually inside 2D, 3D, or Script, despite those being at the top. This points have been brought up a few times. If you have some idea for how to address that, consider making a proposal.
